src/src/PluginStructs/P093_data_struct.cpp

```diff
@@ -342,25 +364,46 @@ void P093_data_struct::applySettings() {
     } else {
       packet[10] = 31 - _wantedSettings.temperature;
     }
+    packet[21] = checkSum(packet, 21);
   }
 
   if (_writeStatus.isDirty(Fan)) {
     packet[11] = _wantedSettings.fan;
     packet[6] |= 0x08;
+    packet[21] = checkSum(packet, 21);
   }
 
   if (_writeStatus.isDirty(Vane)) {
     packet[12] = _wantedSettings.vane;
     packet[6] |= 0x10;
+    packet[21] = checkSum(packet, 21);
```

Comment:
No need to calculate the checksum that often, once, just before sending the packet, is just fine.